Source/Core/Core/CheatSearch.cpp

```diff
@@ -0,0 +1,640 @@
+        r.m_value = current_value.value;
+        r.m_value_state = current_value.translated ?
+                              Cheats::SearchResultValueState::ValueFromVirtualMemory :
+                              Cheats::SearchResultValueState::ValueFromPhysicalMemory;
+        r.m_address = addr;
+      }
+    }
+  });
+  if (error_code == Cheats::SearchErrorCode::Success)
+    return results;
+  return error_code;
+}
+
+Cheats::CheatSearchSessionBase::~CheatSearchSessionBase() = default;
+
+template <typename T>
+Cheats::CheatSearchSession<T>::CheatSearchSession(std::vector<MemoryRange> memory_ranges,
+                                                  PowerPC::RequestedAddressSpace address_space,
+                                                  bool aligned)
+    : m_memory_ranges(std::move(memory_ranges)), m_address_space(address_space), m_aligned(aligned)
+{
+}
+
+template <typename T>
+Cheats::CheatSearchSession<T>::CheatSearchSession(const CheatSearchSession& session) = default;
+
+template <typename T>
+Cheats::CheatSearchSession<T>::CheatSearchSession(CheatSearchSession&& session) = default;
+
+template <typename T>
+Cheats::CheatSearchSession<T>&
+Cheats::CheatSearchSession<T>::operator=(const CheatSearchSession& session) = default;
+
+template <typename T>
+Cheats::CheatSearchSession<T>&
+Cheats::CheatSearchSession<T>::operator=(CheatSearchSession&& session) = default;
+
+template <typename T>
+Cheats::CheatSearchSession<T>::~CheatSearchSession() = default;
+
+template <typename T>
+void Cheats::CheatSearchSession<T>::SetCompareType(CompareType compare_type)
+{
+  m_compare_type = compare_type;
+}
+
+template <typename T>
+void Cheats::CheatSearchSession<T>::SetFilterType(FilterType filter_type)
+{
+  m_filter_type = filter_type;
+}
+
+template <typename T>
+static std::optional<T> ParseValue(const std::string& str)
+{
+  if (str.empty())
+    return std::nullopt;
+
+  T tmp;
+  if (TryParse(str, &tmp))
+    return tmp;
+
+  return std::nullopt;
+}
+
+template <typename T>
+bool Cheats::CheatSearchSession<T>::SetValueFromString(const std::string& value_as_string)
+{
+  m_value = ParseValue<T>(value_as_string);
+  return m_value.has_value();
+}
+
+template <typename T>
+void Cheats::CheatSearchSession<T>::ResetResults()
+{
+  m_first_search_done = false;
+  m_search_results.clear();
+}
+
+template <typename T>
+static std::function<bool(const T& new_value)>
+MakeCompareFunctionForSpecificValue(Cheats::CompareType op, const T& old_value)
+{
+  switch (op)
+  {
+  case Cheats::CompareType::Equal:
+    return [&](const T& new_value) { return new_value == old_value; };
+  case Cheats::CompareType::NotEqual:
+    return [&](const T& new_value) { return new_value != old_value; };
+  case Cheats::CompareType::Less:
+    return [&](const T& new_value) { return new_value < old_value; };
+  case Cheats::CompareType::LessEqual:
+    return [&](const T& new_value) { return new_value <= old_value; };
+  case Cheats::CompareType::More:
```

Comment:
Propose to change from More to Greater, perhaps squeeze in an Or before Equal on the compound types.

Reply:
Yup, good point.